### PR TITLE
Update PGV version

### PR DESCRIPTION
**Pull request by itamarkam**
Commit Message: Update PGV version
Additional Description: This update fails a couple of tests due to Exception error string matching, which is also fixed in this change.
This change is a blocker for [envoyproxy/envoy/pull/15105](https://github.com/envoyproxy/envoy/pull/15105).
Risk Level: Low
Testing: Unit tests
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by itamarkam** [comment]
Note to reviewers - I'm fixing all the tests that failed due to this change, and it's easier for me to find all of them by running all the presubmits here. Once they're done I'll tag specific reviewers and ask for review.

File: api/bazel/repository_locations.bzl
Comment (moderation) [inline api/bazel/repository_locations.bzl:21]: This is not the latest commit. Can we move to the latest?
```
    com_envoyproxy_protoc_gen_validate = dict(
        project_name = "protoc-gen-validate (PGV)",
        project_desc = "protoc plugin to generate polyglot message validators",
        project_url = "https://github.com/envoyproxy/protoc-gen-validate",
        version = "b60bc972ba41dab246dbd0d45bb6a74d82e9ecd7",
        sha256 = "da5f32d925dd07b80a80769a06b0e0f632dff30a656fbd4ffacac2de4a306ffd",
        release_date = "2021-03-10",
        strip_prefix = "protoc-gen-validate-{version}",
        urls = ["https://github.com/envoyproxy/protoc-gen-validate/archive/{version}.tar.gz"],
        use_category = ["api"],
        implied_untracked_deps = [
            "com_github_iancoleman_strcase",
            "com_github_lyft_protoc_gen_star",
            "com_github_spf13_afero",
            "org_golang_google_genproto",
            "org_golang_x_text",
        ],
    ),
```

**Review by moderation — CHANGES_REQUESTED** [review]

File: api/bazel/repository_locations.bzl
Comment (itamarkam) [reply api/bazel/repository_locations.bzl]: Done.

**Review by moderation — DISMISSED** [review]

**Comment by itamarkam** [comment]
Some PGV related tests are failing, for example this one in [linux_x64_tsan](https://dev.azure.com/cncf/envoy/_build/results?buildId=68928&view=logs&j=d1f76054-8f79-554b-6f4a-11d6a63b8e00&t=266e17e3-d213-54b5-deef-0dcee01da137&l=16021).
[linux_x64_asan](https://dev.azure.com/cncf/envoy/_build/results?buildId=68928&view=logs&j=1439b9f7-a348-5b50-b5fe-ea612ea91241&t=37b0a9be-1a71-50b3-594a-4c04a031247d&l=16066) is failing for a different PGV error.
Other linux_x64 tests pass ,for example [linux_x64_gcc](https://dev.azure.com/cncf/envoy/_build/results?buildId=68928&view=logs&j=8bf29878-a4cc-50f7-4e84-2255e6fd4065).

I don't know why there are different failures in different tests, but it doesn't seem like flakiness.
@akonradi Would you mind taking a look?

File: api/bazel/repository_locations.bzl
Comment (moderation) [inline api/bazel/repository_locations.bzl:21]: Another newer commit but happy to do this as a follow up
```
    com_envoyproxy_protoc_gen_validate = dict(
        project_name = "protoc-gen-validate (PGV)",
        project_desc = "protoc plugin to generate polyglot message validators",
        project_url = "https://github.com/envoyproxy/protoc-gen-validate",
        version = "12afc6dcd71f00ea08deee4df84173aed4cc7f97",
        sha256 = "09de7cb0d479fbf3d0aab6f4482bb3f15dfa80c240400fc25226e9f8f1ad490d",
        release_date = "2021-03-12",
        strip_prefix = "protoc-gen-validate-{version}",
        urls = ["https://github.com/envoyproxy/protoc-gen-validate/archive/{version}.tar.gz"],
        use_category = ["api"],
        implied_untracked_deps = [
            "com_github_iancoleman_strcase",
            "com_github_lyft_protoc_gen_star",
            "com_github_spf13_afero",
            "org_golang_google_genproto",
            "org_golang_x_text",
        ],
    ),
```

**Review by moderation — COMMENTED** [review]
I've approved the deps part of the PR but you'll need different reviewers to resolve asan and tsan CI failures

**Comment by akonradi** [comment]
@itamarkam you're right, this isn't flakiness. It looks like we've still got a problem with PGV declaring output files with the wrong paths. Luckily with https://github.com/envoyproxy/protoc-gen-validate/pull/439 we should be able to reproduce this, fix it, and add a test

**Comment by itamarkam** [comment]
> @itamarkam you're right, this isn't flakiness. It looks like we've still got a problem with PGV declaring output files with the wrong paths. Luckily with [envoyproxy/protoc-gen-validate#439](https://github.com/envoyproxy/protoc-gen-validate/pull/439) we should be able to reproduce this, fix it, and add a test

So, there are 3 presubmits that are failing:
1. [asan](https://dev.azure.com/cncf/envoy/_build/results?buildId=69003&view=logs&j=1439b9f7-a348-5b50-b5fe-ea612ea91241&t=37b0a9be-1a71-50b3-594a-4c04a031247d&l=20698) and [tsan](https://dev.azure.com/cncf/envoy/_build/results?buildId=69003&view=logs&j=d1f76054-8f79-554b-6f4a-11d6a63b8e00&t=266e17e3-d213-54b5-deef-0dcee01da137&l=19571) are failing in tests from [filter_example_setup.sh](https://github.com/envoyproxy/envoy/blob/main/ci/filter_example_setup.sh) i.e [this repo](https://github.com/envoyproxy/envoy-filter-example). FWIU it lists Envoy as a submodule and tries to link additional filters to the binary. I can't say I fully understand what's going on there, but my guess is that either something in the protos used there behaves differently than the ones in the PGV fix (so they are not included properly) or that the Envoy version used for these tests doesn't include the PGV update so linking fails (i.e we need to update the Envoy version in the filter example so that the PGV update takes effect there too). @htuch does that make sense?
2. [Compile time options test](https://dev.azure.com/cncf/envoy/_build/results?buildId=69003&view=logs&j=e969334a-0e55-5c18-ac96-8b546753391e&t=b5329a9e-93c7-5412-9f4c-6f8d659e1b57&l=2568) is failing because of a test that builds perfectly well locally, but OTOH it locally (`ci/run_envoy_docker.sh 'ci/do_ci.sh bazel.compile_time_options'`) I see that @envoy//test/common/access_log:access_log_impl_test fails to build due to other PGV dependencies (e.g., "ERROR: /build/tmp/_bazel_envoybuild/400406edc57d332f0b9b805d2b8e33a1/external/envoy_api/envoy/type/metadata/v3/BUILD:7:18: output 'external/envoy_api/envoy/type/metadata/v3/envoy_api/envoy/type/metadata/v3/metadata.pb.validate.h' was not created"). I don't know what's different in the presubmits.

@akonradi This is all blocking the [OT logging PR](https://github.com/envoyproxy/envoy/pull/15105), is there something I can do now to unblock (e.g., patch your PR and root cause it - if so how)?

**Comment by akonradi** [comment]
@itamarkam I just sent https://github.com/envoyproxy/protoc-gen-validate/pull/441 which I think should fix this. You should be able to patch it in by setting the URL in `api/bazel/repository_locations.bzl` for protoc_gen_validate to my fork and using the branch name from the PR as the version.

**Comment by itamarkam** [comment]
> @itamarkam I just sent [envoyproxy/protoc-gen-validate#441](https://github.com/envoyproxy/protoc-gen-validate/pull/441) which I think should fix this. You should be able to patch it in by setting the URL in `api/bazel/repository_locations.bzl` for protoc_gen_validate to my fork and using the branch name from the PR as the version.

@akonradi thanks, it does seem to fix it (I built //source/exe:envoy-static successfully).
I'll update it once your change is merged.

File: test/extensions/filters/network/mongo_proxy/config_test.cc
Comment (akonradi) [inline test/extensions/filters/network/mongo_proxy/config_test.cc:98]: Nit: no need for a raw string literal. Same below.

**Review by akonradi — CHANGES_REQUESTED** [review]
Minor nit otherwise LGTM

File: test/extensions/filters/network/mongo_proxy/config_test.cc
Comment (itamarkam) [reply test/extensions/filters/network/mongo_proxy/config_test.cc]: Done.

Comment (akonradi) [reply test/extensions/filters/network/mongo_proxy/config_test.cc]: I suspect the parens are going to make CI fail

Comment (itamarkam) [reply test/extensions/filters/network/mongo_proxy/config_test.cc]: Yup.
I reverted the change, and then just removed the R and parens (don't know why I didn't do that the first place).
Either way, now done.

**Review by akonradi — APPROVED** [review]
Thanks!

File: api/bazel/repository_locations.bzl
Comment (moderation) [inline api/bazel/repository_locations.bzl:21]: We've alternated between releases and commits with PGV but @akonradi cut `v0.5.0` of PGV yesterday - can we switch to that? https://github.com/envoyproxy/protoc-gen-validate/releases/tag/v0.5.0. From a dependency policy point of view we prefer releases.
```python
    com_envoyproxy_protoc_gen_validate = dict(
        project_name = "protoc-gen-validate (PGV)",
        project_desc = "protoc plugin to generate polyglot message validators",
        project_url = "https://github.com/envoyproxy/protoc-gen-validate",
        version = "0.5.0",
        sha256 = "b3f51da19196174fca07d947adcadf478d6825ea35d3261c9562532f38f0ea94",
        release_date = "2021-03-15",
        strip_prefix = "protoc-gen-validate-{version}",
        urls = ["https://github.com/envoyproxy/protoc-gen-validate/archive/v{version}.tar.gz"],
        use_category = ["api"],
        implied_untracked_deps = [
            "com_github_iancoleman_strcase",
            "com_github_lyft_protoc_gen_star",
            "com_github_spf13_afero",
            "org_golang_google_genproto",
            "org_golang_x_text",
        ],
    ),
```

Comment (itamarkam) [reply api/bazel/repository_locations.bzl:21]: Unfortunately, this release [doesn't include the fix to some of the presubmit failures](https://github.com/envoyproxy/protoc-gen-validate/compare/v0.5.0...main), specifically [this commit](https://github.com/envoyproxy/protoc-gen-validate/commit/9db8e779b46119eef7d67e3d1120c118d593611c).

**Review by yanavlasov — APPROVED** [review]

**Pull request merged**
